sample acclerometer(source yt)/MainActivity.java

```diff
@@ -0,0 +1,68 @@
+package com.example.myapplicationas;
+
+import androidx.appcompat.app.AppCompatActivity;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.Bundle;
+import android.widget.TextView;
+/*CODE FROM
+* YT-Android Tutorial Spot
+* video link-https://youtu.be/LsWJipo4knk*/
+public class MainActivity extends AppCompatActivity implements SensorEventListener {
+/*only the abv lines i understand*/
+
+    private TextView textView;
+    private SensorManager sensorManager;
+    private Sensor sensor;
+
+
+
+
+    /*i dnt know what type of variable or objects are created abv
+    * i am only familiar with the one where we give data type first
+    *  idk how this works */
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        /*idk whats going in the brackets aftr on create
+        *method are there parameters in brackets or something else idk..
+        * what topic in java is it?*/
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);

```

Comment:
Ok so the topic here is called “inheritance”, you are working with a class called MainActivity ok, this class extends AppCompatActivity, this means that it ‘inherits’ all the methods of the class it is extending. For instance super.onCreate and setContentView are two functions that were declared in AppCompatActivity and you can use them here because you are extending that class. If you really want to figure out what they do (even if the name is pretty clear) you can control click and go at the definition of the ‘parent’ class (that’s how it’s classed when you inherit from) and check what these methods do